packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -981,4 +1062,89 @@ class AndroidCameraCameraX extends CameraPlatform {
     return proxy.createQualitySelector(
         videoQuality: videoQuality, fallbackStrategy: fallbackStrategy);
   }
+
+  // Methods for configuring auto-focus and auto-exposure:
+
+  /// Starts a foucs and metering action.
+  ///
+  /// This method will merge the current action's metering points with that
+  /// specified such that only one auto-exposure metering point and one
+  /// auto-focus metering point will be part of the action. This method would
+  /// also accommodate at most one auto-white balance metering point, but that
+  /// mode is currently unused in this plugin.
+  ///
+  /// Thus, if [meteringPoint] is non-null, this action includes:
+  ///   * metering points and their modes previously added to
+  ///     [currentFocusMeteringAction] that do not share a metering mode with
+  ///     [meteringPoint] and
+  ///   * [meteringPoint] with the specified [meteringMode].
+  /// If [meteringPoint] is null, this action includes only metering points and
+  /// their modes previously added to [currentFocusMeteringAction] that do not
+  /// share a metering mode with [meteringPoint]. If there are no such metering
+  /// points, then the previously enabled focus and metering actions will be
+  /// canceled.
+  Future<void> _startFocusAndMeteringFor(
+      {required Point<double>? meteringPoint,
+      required int meteringMode}) async {
+    if (meteringPoint == null) {
+      // Try to clear any metering point from previous action with the specified
+      // meteringMode.
+      if (currentFocusMeteringAction == null) {
+        // Attempting to clear a metering point from a previous action, but no
+        // such action exists.
+        return;
+      }
+
+      // Remove metering point with specified meteringMode from current focus
+      // and metering action, as only one focus or exposure point may be set
+      // at once in this plugin.
```

Comment:
Why do we have this restriction? Also, do you happen to know what the use case is for having multiple focus or exposure points set? I just want to better understand why we are taking a different approach than CameraX and what the impact is

Reply:
This was an older design decision made for the plugin, but I'm not entirely why. I suppose you may want to balance the exposure based on two focal points or have multiple points of focus. CameraX does warn [in the docs](https://developer.android.com/reference/androidx/camera/core/CameraControl#startFocusAndMetering(androidx.camera.core.FocusMeteringAction)) that this may not be available for all devices, though. Perhaps it's a newer feature and that's why it wasn't supported in the past.